webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -293,6 +363,84 @@ export class MonacoService {
     });
   }
 
+  /* save a dataset, not a file */
+  saveDataset(fileContext: ProjectContext, fileDirectory?: string): Observable<void> {
+    return new Observable((obs) => {
+      this.log.warn(`saveDataset name `, fileContext.model.fileName, ` directory `, fileDirectory );
+      
+      /* always save it, for now.  Fix it properly later */
+      this.editorControl.saveBufferToDataset(fileContext, fileContext.model.fileName).subscribe(() => obs.next());
+      return;      
```

Comment:
I'm a bit confused about this method. Everything after this return doesn't get run. The return will end the method. Did you mean to put it inside the subscribe, or is the other majority of this method useless?